File: backend/app/schemas/history_chart.py
from pydantic import BaseModel
from datetime import date
from decimal import Decimal
from typing import List

class PortfolioPoint(BaseModel):
    day: date
    total_value: Decimal

    class Config:
        from_attributes = True

class PortfolioHistoryResponse(BaseModel):
    history: List[PortfolioPoint]

File: frontend/src/types/history_chart.ts
export interface PortfolioPoint {
  date: string;
  total_value: string;
  capital_invertido: string;
}

export interface PortfolioHistoryResponse {
  history: PortfolioPoint[];
}

export interface ChartDataPoint {
  day: string;
  total_value: number; 
  capital_invertido: number;
  profit: number;
  displayDate: string;
}